src/shell-window-tracker.c

```diff
@@ -739,6 +830,10 @@ shell_window_tracker_init (ShellWindowTracker *self)
 
   self->launched_pid_to_app = g_hash_table_new_full (NULL, NULL, NULL, (GDestroyNotify) g_object_unref);
 
+  self->builder_to_app = g_hash_table_new_full (NULL, NULL, NULL, (GDestroyNotify) g_object_unref);
```

Comment:
Nothing ever refs what we inserts into this table, so we should not have `g_object_unref()` as the destroy function.
However, you should make sure to remove the window also from this table when it's destroyed.